modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -1597,10 +1597,10 @@ protected KBArticle getNextKBArticle(
 	}
 
 	protected KBArticle[] getPreviousAndNextKBArticles(KBArticle kbArticle) {
-		List<KBArticle> kbArticles = kbArticlePersistence.findByG_P_S(
+		List<KBArticle> kbArticles = kbArticlePersistence.findByG_P_M(
 			kbArticle.getGroupId(), kbArticle.getParentResourcePrimKey(),
-			WorkflowConstants.STATUS_APPROVED, QueryUtil.ALL_POS,
-			QueryUtil.ALL_POS, new KBArticlePriorityComparator(true));
+			true, QueryUtil.ALL_POS, QueryUtil.ALL_POS,
```

Comment:
We should do the same change in the above method getNextKBArticle and the below method getPreviousKBArticle